src/mpid/ch4/netmod/ucx/ucx_datatype.h

```diff
@@ -84,6 +84,9 @@ static inline ucs_status_t MPIDI_UCX_Unpack(void *state, size_t offset, const vo
 
     MPID_Segment_unpack(pack_state->segment_ptr, offset, &last, (void *) src);
 
+    if(unlikely (last == 0)){
```

Comment:
Is this the right test? `last` is an in/out parameter. We need to save the input value in order to tell the same value was actually unpacked.